astropy/modeling/separable.py

```diff
@@ -301,7 +301,9 @@ def _separable(transform):
         An array of shape (transform.n_outputs,) of boolean type
         Each element represents the separablity of the corresponding output.
     """
-    if isinstance(transform, CompoundModel):
+    if (transform_matrix := transform._calculate_separability_matrix()) is not NotImplemented:
```

Comment:
Can we use walrus in the backport? I don't remember.

Reply:
Looks like it: https://github.com/astropy/astropy/blob/v5.0.x/setup.cfg#L32